From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Wolfgang Bumiller <w.bumiller@proxmox.com>
Date: Sun, 5 Apr 2020 16:12:45 +0200
Subject: [PATCH] get the right path in get_cgroup command

Signed-off-by: Wolfgang Bumiller <w.bumiller@proxmox.com>
---
 src/lxc/commands.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/lxc/commands.c b/src/lxc/commands.c
index 44714f9ba..d735b5ff6 100644
--- a/src/lxc/commands.c
+++ b/src/lxc/commands.c
@@ -592,8 +592,8 @@ static int lxc_cmd_get_cgroup_callback_do(int fd, struct lxc_cmd_req *req,
 		reqdata = NULL;
 	}
 
-	get_fn = (limiting_cgroup ? cgroup_ops->get_cgroup
-				  : cgroup_ops->get_limiting_cgroup);
+	get_fn = (limiting_cgroup ? cgroup_ops->get_limiting_cgroup
+				  : cgroup_ops->get_cgroup);
 
 	path = get_fn(cgroup_ops, reqdata);
 
