From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Wolfgang Bumiller <w.bumiller@proxmox.com>
Date: Wed, 18 Jun 2025 10:08:22 +0200
Subject: [PATCH 1/2] apparmor: allow lxc-start to create user namespaces

Signed-off-by: Wolfgang Bumiller <w.bumiller@proxmox.com>
---
 config/apparmor/abstractions/start-container.in | 1 +
 1 file changed, 1 insertion(+)

diff --git a/config/apparmor/abstractions/start-container.in b/config/apparmor/abstractions/start-container.in
index 20c852925..799bd141c 100644
--- a/config/apparmor/abstractions/start-container.in
+++ b/config/apparmor/abstractions/start-container.in
@@ -7,6 +7,7 @@
   dbus,
   signal,
   ptrace,
+  userns,
 
   # currently blocked by apparmor bug
   mount -> /usr/lib*/*/lxc/{**,},
-- 
2.47.2

