From 2e62657e65daad24f4c11744ce133e8c4d75df02 Mon Sep 17 00:00:00 2001
From: Wolfgang Bumiller <w.bumiller@proxmox.com>
Date: Fri, 12 Aug 2016 14:55:42 +0200
Subject: [PATCH 3/3] doc: add --rcfile to common opts

Signed-off-by: Wolfgang Bumiller <w.bumiller@proxmox.com>
---
 doc/common_options.sgml.in | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/doc/common_options.sgml.in b/doc/common_options.sgml.in
index 38783dd..978c0ba 100644
--- a/doc/common_options.sgml.in
+++ b/doc/common_options.sgml.in
@@ -107,6 +107,21 @@ Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
     </varlistentry>
 
     <varlistentry>
+      <term><option>--rcfile=<replaceable>FILE</replaceable></option></term>
+      <listitem>
+	<para>
+	  Specify the configuration file to configure the virtualization
+	  and isolation functionalities for the container.
+	</para>
+	<para>
+	 This configuration file if present will be used even if there is
+	 already a configuration file present in the previously created
+	 container (via lxc-create).
+	</para>
+      </listitem>
+    </varlistentry>
+
+    <varlistentry>
       <term><option>--version</option></term>
       <listitem>
         <para>
-- 
2.1.4

