From c1c1e55305a06786ee3dd938e421ca413db73dd1 Mon Sep 17 00:00:00 2001
From: Wolfgang Bumiller <w.bumiller@proxmox.com>
Date: Wed, 6 Sep 2017 11:51:03 +0200
Subject: [PATCH 9/9] network: add missing checks for empty links

Signed-off-by: Wolfgang Bumiller <w.bumiller@proxmox.com>
---
 src/lxc/network.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/lxc/network.c b/src/lxc/network.c
index 3c0597c7..0ad42318 100644
--- a/src/lxc/network.c
+++ b/src/lxc/network.c
@@ -2355,7 +2355,7 @@ bool lxc_delete_network_unpriv(struct lxc_handler *handler)
 		if (netdev->type != LXC_NET_VETH)
 			continue;
 
-		if (!is_ovs_bridge(netdev->link))
+		if (netdev->link[0] == '\0' || !is_ovs_bridge(netdev->link))
 			continue;
 
 		if (netdev->priv.veth_attr.pair[0] != '\0')
@@ -2564,7 +2564,7 @@ bool lxc_delete_network_priv(struct lxc_handler *handler)
 		}
 		INFO("Removed interface \"%s\" from \"%s\"", hostveth, netdev->link);
 
-		if (!is_ovs_bridge(netdev->link)) {
+		if (netdev->link[0] == '\0' || !is_ovs_bridge(netdev->link)) {
 			netdev->priv.veth_attr.veth1[0] = '\0';
 			continue;
 		}
-- 
2.11.0

