From b50f5a815258541b6da0c772388d93ea303a4f3d Mon Sep 17 00:00:00 2001
From: Christian Brauner <christian.brauner@ubuntu.com>
Date: Sat, 11 Nov 2017 23:05:33 +0100
Subject: [PATCH 13/13] console: use correct escape sequence check

Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>
---
 src/lxc/console.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/lxc/console.c b/src/lxc/console.c
index 8f6203f0..72c01b19 100644
--- a/src/lxc/console.c
+++ b/src/lxc/console.c
@@ -616,7 +616,7 @@ int lxc_console_cb_tty_stdin(int fd, uint32_t events, void *cbdata,
 	if (lxc_read_nointr(ts->stdinfd, &c, 1) <= 0)
 		return 1;
 
-	if (ts->escape != -1) {
+	if (ts->escape >= 1) {
 		/* we want to exit the console with Ctrl+a q */
 		if (c == ts->escape && !ts->saw_escape) {
 			ts->saw_escape = 1;
-- 
2.11.0

