From b68958e110feed24a632d9e157d235e0a1980a17 Mon Sep 17 00:00:00 2001
From: Wolfgang Bumiller <w.bumiller@proxmox.com>
Date: Fri, 30 Jun 2017 10:51:19 +0200
Subject: [PATCH 15/15] fix segfault in lxc-attach

Signed-off-by: Wolfgang Bumiller <w.bumiller@proxmox.com>
---
 src/lxc/attach.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/lxc/attach.c b/src/lxc/attach.c
index 6fdc6e8a..1dff59ca 100644
--- a/src/lxc/attach.c
+++ b/src/lxc/attach.c
@@ -855,7 +855,10 @@ int lxc_attach(const char* name, const char* lxcpath, lxc_attach_exec_t exec_fun
 		}
 
 		/* Setup resource limits */
-		if (!lxc_list_empty(&init_ctx->container->lxc_conf->limits) && setup_resource_limits(&init_ctx->container->lxc_conf->limits, pid)) {
+		if (init_ctx->container &&
+		    init_ctx->container->lxc_conf &&
+		    !lxc_list_empty(&init_ctx->container->lxc_conf->limits)
+		    && setup_resource_limits(&init_ctx->container->lxc_conf->limits, pid)) {
 			goto on_error;
 		}
 
-- 
2.11.0

