diff --git a/src/lxc/conf.c b/src/lxc/conf.c
index 0e3421b..91c90b1 100644
--- a/src/lxc/conf.c
+++ b/src/lxc/conf.c
@@ -3083,9 +3083,13 @@ void lxc_delete_network(struct lxc_handler *handler)
 		 * namespace is destroyed but in case we did not moved the
 		 * interface to the network namespace, we have to destroy it
 		 */
-		if (netdev->ifindex != 0 &&
-		    lxc_netdev_delete_by_index(netdev->ifindex))
-			WARN("failed to remove interface '%s'", netdev->name);
+		if (netdev->ifindex != 0) {
+			int err = lxc_netdev_delete_by_index(netdev->ifindex);
+			if (err) {
+				WARN("failed to remove interface '%s': %i: %s", netdev->name,
+				     err, strerror(-err));
+			}
+		}
 	}
 }
 
