Index: new/src/lxc/lxccontainer.c
===================================================================
--- new.orig/src/lxc/lxccontainer.c
+++ new/src/lxc/lxccontainer.c
@@ -924,10 +924,10 @@ static struct bdev *do_bdev_create(struc
 		dest = alloca(len);
 		ret = snprintf(dest, len, "%s", rpath);
 	} else {
-		const char *lxcpath = do_lxcapi_get_config_path(c);
-		len = strlen(c->name) + strlen(lxcpath) + 9;
+		const char *rootfs_path = "/var/lib/lxc";
+		len = strlen(c->name) + strlen(rootfs_path) + 9;
 		dest = alloca(len);
-		ret = snprintf(dest, len, "%s/%s/rootfs", lxcpath, c->name);
+		ret = snprintf(dest, len, "%s/%s/rootfs", rootfs_path, c->name);
 	}
 	if (ret < 0 || ret >= len)
 		return NULL;
