Index: new/src/lxc/namespace.h
===================================================================
--- new.orig/src/lxc/namespace.h
+++ new/src/lxc/namespace.h
@@ -23,10 +23,12 @@
 #ifndef __LXC_NAMESPACE_H
 #define __LXC_NAMESPACE_H
 
+#include "config.h"
+
 #include <sys/syscall.h>
+#include <linux/sched.h>
 #include <sched.h>
 
-#include "config.h"
 
 #ifndef CLONE_FS
 #  define CLONE_FS                0x00000200
