From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: C0deAi <benjamin.bales@assrc.us>
Date: Fri, 6 Apr 2018 12:46:53 -0400
Subject: [PATCH] storage/rsync: free memory on error

Closes #2262.

Signed-off-by: C0deAi <techsupport@mycode.ai>
Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>
---
 src/lxc/storage/rsync.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/lxc/storage/rsync.c b/src/lxc/storage/rsync.c
index c3080a56..83871ae8 100644
--- a/src/lxc/storage/rsync.c
+++ b/src/lxc/storage/rsync.c
@@ -73,8 +73,10 @@ int lxc_rsync_exec(const char *src, const char *dest)
 		return -1;
 
 	ret = snprintf(s, l, "%s", src);
-	if (ret < 0 || (size_t)ret >= l)
+	if (ret < 0 || (size_t)ret >= l) {
+		free(s);
 		return -1;
+	}
 
 	s[l - 2] = '/';
 	s[l - 1] = '\0';
-- 
2.11.0

